internal/store/store.go

```diff
@@ -883,6 +884,19 @@ func (s Store) GetIngressClassParametersV1Alpha1() (*kongv1alpha1.IngressClassPa
 	return params.(*kongv1alpha1.IngressClassParameters), nil
 }
 
+// GetGateway returns gateway resource having specified namespace and name.
+func (s Store) GetGateway(namespace string, name string) (*gatewayv1beta1.Gateway, error) {
```

Comment:
I would much prefer if we used [NamespacedName](https://pkg.go.dev/k8s.io/apimachinery/pkg/types#NamespacedName) for functions like this across our codebase.

Reply:
If we want to use `NamespacedName` here for all types, it would be a quite large change. So I prefer to do that change in a separate PR. WDYT is the new method should use in this PR, switch to `NamespacedName` in advance, or keep in same as the other methods?

Reply:
Do we get anything in particular out of using it? There's a standardized string function we could use for the key but that's not huge. I'd say keep it consistent with the other signatures in here and convert them all at once separately if we do want to refactor it.

Reply:
kept this as the existing methods.